src/core/engine/processors/java_etl_mongo_order_status_histories_processor.py

```diff
@@ -0,0 +1,63 @@
+from bson import ObjectId
+
+from src.constants.etl_status import EtlStatus, Service
+from src.core.engine.processors.abstract_processor import AbstractProcessor
+from src.core.engine.processors.etl.mongo_order_status_history_distribution_etl import (
+    UNASSIGNED_ORDER_STATUS_HISTORIES_LIMIT,
+)
+from src.grpc.clients.ea_restaurant_java_etl_grpc_client import (
+    EaRestaurantJavaEtlGrpcClient,
+)
+from src.lib.repositories.impl_no_sql.order_status_history_repository_impl import (
+    OrderStatusHistoryRepositoryImpl as MongoOrderStatusHistoryRepository,
+)
+
+UNASSIGNED_ORDER_STATUS_HISTORIES_LIMIT = 1
+
+
+class JavaEtlMongoOrderStatusHistoriesProcessor(AbstractProcessor):
+    def __init__(self, app_processor_config, app_context=None):
+        super().__init__(
+            app_processor_config=app_processor_config, app_context=app_context
+        )
+
+        self.mongo_order_status_history_repository = None
+        self.ea_restaurant_java_etl_grpc_client = None
+
+    def set_app_context(self, app_context):
+        ioc = app_context.ioc
+        self.mongo_order_status_history_repository = ioc.get(
+            MongoOrderStatusHistoryRepository
+        )
+        self.ea_restaurant_java_etl_grpc_client = ioc.get(EaRestaurantJavaEtlGrpcClient)
+
+    def process(self, delta_time):
+        mongo_order_status_histories = (
+            self.get_unprocessed_java_etl_mongo_order_status_histories()
+        )
+
+        if mongo_order_status_histories:
+            self.send_mongo_order_status_histories_to_java_etl(
+                mongo_order_status_histories
+            )
+
+    def get_unprocessed_java_etl_mongo_order_status_histories(self):
+        mongo_order_status_histories = self.mongo_order_status_history_repository.get_order_status_histories_by_service(
+            Service.JAVA_ETL,
+            EtlStatus.UNPROCESSED,
+            limit=UNASSIGNED_ORDER_STATUS_HISTORIES_LIMIT,
+        )
+        return list(mongo_order_status_histories)
```

Comment:
I feel you will need to do this every time you request any data to mongo. What do you think about moving this transformation inside `self.mongo_order_status_history_repository.get_order_status_histories_by_service`?

If you do that, thenm you wouldn't need to `def get_unprocessed_java_etl_mongo_order_status_histories(self):`